lib/SILOptimizer/Utils/OwnershipOptUtils.cpp

```diff
@@ -201,9 +201,15 @@ CopyValueInst *OwnershipLifetimeExtender::copyAndExtendForNonLifetimeEndingRAUW(
   auto opRange = makeUserRange(range);
   ValueLifetimeAnalysis lifetimeAnalysis(copy, opRange);
   ValueLifetimeAnalysis::Frontier frontier;
-  bool result = lifetimeAnalysis.computeFrontier(
-      frontier, ValueLifetimeAnalysis::DontModifyCFG, &ctx.deBlocks);
-  assert(result);
+  bool noCriticalEdges = lifetimeAnalysis.computeFrontier(
```

Comment:
I think this is the wrong way to do this. Why is this needed?

Reply:
Specifically, if it is needed due to the terminator issue we spoke about before, I think that Andy's recent change adding UnownedInstantaneousUse makes that not necessary for unowned things

Reply:
We may have to split the edges while computing frontiers if we have a use on the terminator instruction.

Here's an example:

```
sil [ossa] @struct_extract_test4 : $@convention(thin) (@guaranteed StructWithEnum2) -> () {
bb0(%0 : @guaranteed $StructWithEnum2):
 %1 = struct_extract %0 : $StructWithEnum2, #StructWithEnum2.val
 %copy1 = copy_value %1 : $FakeOptional2
 br bb4

bb4:
 %2 = struct_extract %0 : $StructWithEnum2, #StructWithEnum2.val
 %copy2 = copy_value %2 : $FakeOptional2
 switch_enum %2 : $FakeOptional2, case #FakeOptional2.some1!enumelt:bb5, case #FakeOptional2.some2!enumelt:bb6

bb5(%arg4 : $UInt):
 br bb7(%arg4 : $UInt) --> needs split here

bb6(%arg5 : @guaranteed $Klass):
 %4 = unchecked_trivial_bit_cast %arg5 : $Klass to $UInt
 br bb7(%4 : $UInt)

bb7(%arg6 : $UInt):
 destroy_value %copy2 : $FakeOptional2
 br bb8

bb8:
 destroy_value %copy1 : $FakeOptional2
 %res = tuple ()
 return %res : $()
}
```
Here we are CSE'ing the `struct_extract` at %2 with %1. For this we will have to split the edges at bb5 to get a frontier after the branch.

Reply:
You are right, this was not needed. It turned out to be a bug where the ownership rauw was considering transitive uses of none ownership as well.